### PR TITLE
Feature : Add cancellable IslandKickEvent

### PR DESCRIPTION
**Pull request by NewZAZ**
Add possibility to cancel IslandKickEvent !

### CODE REVIEW

**Comment by OmerBenGera** [comment]
There are commits not related to the feature at all.
Make sure you only push related changes

**Comment by NewZAZ** [comment]
Oh sorry i didn't see this i fix this

**Comment by NewZAZ** [comment]
@OmerBenGera it's fixed :)

File: src/main/java/com/bgsoftware/superiorskyblock/island/IslandUtils.java [hunk 1]
```diff
@@ -218,7 +218,8 @@ public static void handleKickPlayer(SuperiorPlayer caller, Island island, Superi
     }
 
     public static void handleKickPlayer(SuperiorPlayer caller, String callerName, Island island, SuperiorPlayer target) {
-        plugin.getEventsBus().callIslandKickEvent(caller, target, island);
+        if(plugin.getEventsBus().callIslandKickEvent(caller, target, island))
```
Comment (OmerBenGera) [inline src/main/java/com/bgsoftware/superiorskyblock/island/IslandUtils.java]: callIslandKickEvent will return whether the event was successfully called.
Therefore, you want to negate the check - it should return when the method returns false.

File: src/main/java/com/bgsoftware/superiorskyblock/mission/MissionsManagerImpl.java [hunk 2]
```diff
@@ -330,7 +330,6 @@ public void saveMissionsData() {
 
         if (!dataFolder.exists())
             dataFolder.mkdirs();
-
```
Comment (OmerBenGera) [inline src/main/java/com/bgsoftware/superiorskyblock/mission/MissionsManagerImpl.java]: Not relevant to the feature?

**Review by OmerBenGera — CHANGES_REQUESTED** [review]

**Comment by NewZAZ** [comment]
That's good @OmerBenGera

**Comment by SwartZCoding** [comment]
Nice idea to add asap

**Comment by LuluxClient** [comment]
No 😨

**Comment by OmerBenGera** [comment]
Why is the MissionsManagerImpl was edited? How is it related to the PR?
Make a proper PR - Create a new branch from dev, make your changes, then submit them.
Only changes related to the PR should be pushed - otherwise, I will reject the PR.

**Pull request closed without merging**